### PR TITLE
Fix issue #1261 WorkbookWriter sheet.protect() function doesn't exist

### PR DESCRIPTION
**Pull request by rheidari**
## Changes

- Added `protect` method to `worksheet-writer.js` (copied from `worksheet.js`)
- Fix for #1261

### CODE REVIEW

File: lib/stream/xlsx/worksheet-writer.js [hunk 1]
```diff
@@ -450,6 +451,39 @@ class WorksheetWriter {
     return this._background && this._background.imageId;
   }
 
+  // =========================================================================
+  // Worksheet Protection
+  protect(password, options) {
+    // TODO: make this function truly async
+    // perhaps marshal to worker thread or something
+    return new Promise(resolve => {
```
Comment (Alanscut) [inline lib/stream/xlsx/worksheet-writer.js:459]: Whether this attribute should be declared in `constructor ()`.
```javascript
    // worksheet protection
    this.sheetProtection = null;
```

Comment (rheidari) [reply lib/stream/xlsx/worksheet-writer.js:461]: This has been added to the constructor

**Comment by alubbe** [comment]
Please merge the current master into your branch

**Comment by alubbe** [comment]
The implementation looks good, but I think we need a better test here. Ideally, write a file with a protection and then show that modifying it produces an error

**Comment by rheidari** [comment]
> @rheidari, have you possibility to made requested changes?

Apologies for the delay, I've updated the test!

**Comment by raeesaa** [comment]
Any updates on this issue?

**Review by Alanscut — APPROVED** [review]

File: lib/stream/xlsx/worksheet-writer.js (hunk 1 above)
Comment (VikasAnand354) [reply lib/stream/xlsx/worksheet-writer.js:485]: Does this really work ? I set sheet protection to true it doesn't lock a particular cell

**Pull request merged**